integration-tests/cli/product-json-to-xlsx.it.js

```diff
@@ -0,0 +1,885 @@
+          const columns = [
+            '_published',
+            '_hasStagedChanges',
+            'key',
+            'productType',
+            'variantId',
+            'sku',
+            'text-attribute',
+            'another-text-attribute',
+          ]
+          products.forEach(product => {
+            const productColumns = Object.keys(product)
+            expect(productColumns).toEqual(columns)
+          })
+        })
+
+        it('should contain variants custom attributes', () => {
+          // From first product
+          expect(products[0]).toEqual(
+            expect.objectContaining({ 'text-attribute': 'Master Var attr' })
+          )
+          expect(products[0]).toEqual(
+            expect.objectContaining({ 'another-text-attribute': undefined })
+          )
+          expect(products[1]).toEqual(
+            expect.objectContaining({ 'text-attribute': 'First Var attr' })
+          )
+          expect(products[1]).toEqual(
+            expect.objectContaining({ 'another-text-attribute': undefined })
+          )
+          expect(products[2]).toEqual(
+            expect.objectContaining({ 'text-attribute': 'Second Var attr' })
+          )
+          expect(products[2]).toEqual(
+            expect.objectContaining({ 'another-text-attribute': undefined })
+          )
+
+          // From second product
+          expect(products[3]).toEqual(
+            expect.objectContaining({ 'text-attribute': undefined })
+          )
+          expect(products[3]).toEqual(
+            expect.objectContaining({
+              'another-text-attribute': 'Another Master Var attr',
+            })
+          )
+          expect(products[4]).toEqual(
+            expect.objectContaining({ 'text-attribute': undefined })
+          )
+          expect(products[4]).toEqual(
+            expect.objectContaining({
+              'another-text-attribute': 'Another First Var attr',
+            })
+          )
+        })
+      })
+    })
+
+    describe('From JSON file', () => {
+      let productsJsonFile
+      beforeAll(async () => {
+        productsJsonFile = tmp.fileSync({ postfix: '.json' }).name
+        const cmd = `${exporter} -p ${projectKey} -s -o ${productsJsonFile}`
+        const [stdout, stderr] = await exec(cmd)
+
+        expect(stdout).toBeTruthy()
+        expect(stderr).toBeFalsy()
+      }, 10000)
+
+      describe('WITHOUT HEADERS::should write products to `zip` file', () => {
+        let product1
+        let product2
+        const fileNames = []
+
+        beforeAll(async done => {
+          const zipFile = tmp.fileSync({ postfix: '.zip' }).name
+
+          // Map products from a JSON file to archived XLSX files
+          const [parseStdout, parseStderr] = await exec(
+            `${binPath} -p ${projectKey} -i ${productsJsonFile} --referenceCategoryBy namedPath --fillAllRows -o ${zipFile}`
+          )
+          expect(parseStdout).toBeTruthy()
```

Comment:
same question

Reply:
This came from product-json-to-csv .. I removed it here 233983c from xlsx and here 27c2288 from csv mapper